yaas/blueprint.py

```diff
@@ -1,137 +1,54 @@
 # coding: utf-8
 
+""" Interact with Ambari blueprints. """
+
 from __future__ import absolute_import
 from __future__ import print_function
 
-import argparse
-import inspect
 import json
 import pprint
 import requests
 import sys
 
-from . import common
-
-
-def command(parser, args):
-    """ Interact with Ambari blueprints. """
-    subparsers = parser.add_subparsers(dest='subcommand')
-    subcommands = {
-        'add': _add,
-        'list': _list,
-        'remove': _remove,
-        'show': _show,
-        }
-    for name, fn in subcommands.items():
-        subparsers.add_parser(name, help=inspect.getdoc(fn))
-    subargs, extra = parser.parse_known_args(args)
-    subcommands[subargs.subcommand](
-        subparsers.choices[subargs.subcommand],
-        extra)
+from . import config
+from . import utils
 
-
-def _add(parser, args):
+def add(blueprint, blueprint_name, validate_topology=True):
     """ Add a blueprint to the Ambari server. """
-    parser.add_argument(
-        '--validate-topology',
-        help="Specify whether to validate a blueprint on upload.")
-    parser.add_argument(
-        'blueprint',
-        nargs='*',
-        help="Specify a blueprint to add to the Ambari server.")
-    subargs = parser.parse_args(args)
-    blueprints = [] if len(subargs.blueprint) > 0 else [sys.stdin.read()]
-    for blueprint_file in subargs.blueprint:
-        with open(blueprint_file, 'r') as f:
-            blueprints.append(f.read())
-    for blueprint in blueprints:
-        blueprint = json.loads(blueprint)
-        response = requests.post(
-            common.href(
-                '/api/v1/blueprints/{name}'.format(
-                    name=blueprint['Blueprints']['blueprint_name'])),
-            data=json.dumps(blueprint),  # `json=blueprint` would be better, but it doesn't work!
-            params={'validate_topology': subargs.validate_topology},
-            **common.requests_opts())
-        response.raise_for_status()
+    response = requests.post(
+        config.href(
+            '/api/v1/blueprints/{name}'.format(name=blueprint_name)),
+        data=json.dumps(blueprint),  # `json=blueprint` would be better, but it doesn't work!
+        params={'validate_topology': validate_topology},
+        **config.requests_opts())
 
+    response.raise_for_status()
```

Comment:
Not quite (altho that is interesting)...
More like:
``` python
def add(...):
    return requests.post(
        config.href(
            '/api/v1/blueprints/{name}'.format(name=blueprint_name)),
        data=json.dumps(blueprint),  # `json=blueprint` would be better, but it doesn't work!
        params={'validate_topology': validate_topology},
        **config.requests_opts())
```
and let the caller do whatever they want with the response object.